libraries/helpers.rb

```diff
@@ -432,3 +432,51 @@ def overwrite_nfs_template?
     node['platform'] == 'centos' && node['platform_version'].to_i == 8
   ].any?
 end
+
+def setup_munge_head_node
+  # Generate munge key
+  bash 'generate_munge_key' do
+    user 'root'
+    group 'root'
+    cwd '/tmp'
+    code <<-MUNGE_KEY
+      set -e
+      # Generates munge key in /etc/munge/munge.key
+      sudo -u munge /usr/sbin/mungekey --verbose
+      # Enforce correct permission on the key
+      chmod 0600 /etc/munge/munge.key
+      chown munge:munge /etc/munge/munge.key
+      # Copy key to shared dir, this tmp key is owned by root
+      cp /etc/munge/munge.key /home/#{node['cfncluster']['cfn_cluster_user']}/.munge.key
```

Comment:
shall we copy with `-p` so that we preserve the munge ownership and don't need to enforce in compute nodes again?or actually we could run this bash script directly as munge user

Reply:
also why not using the home of the munge user to share the key?

Reply:
> shall we copy with -p so that we preserve the munge ownership and don't need to enforce in compute nodes again?or actually we could run this bash script directly as munge user

Definitely yes for `-p`, I will try and see if I can run the script as munge user, if not I will keep running as root

> also why not using the home of the munge user to share the key?

I just did it because we don't have a home directory for munge user, but we can create one. Currently we have:
```
$ ls /home
dcvextauth  sgeadmin  slurm  ubuntu
```